net-libs/libmicrohttpd/libmicrohttpd-0.9.73.ebuild

```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit multilib-minimal
+
+MY_P="${P/_/}"
+
+DESCRIPTION="Small C library to run an HTTP server as part of another application"
+HOMEPAGE="https://www.gnu.org/software/libmicrohttpd/"
+SRC_URI="mirror://gnu/${PN}/${MY_P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0/12"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86"
+IUSE="+epoll ssl static-libs test thread-names"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	ssl? ( virtual/pkgconfig )
+	test? ( virtual/pkgconfig )
+	"
+
+RDEPEND="ssl? ( >net-libs/gnutls-2.12.20:= )"
+
+DEPEND="${RDEPEND}
+	test? ( net-misc/curl[ssl?] )
```

Comment:
This applies only for testing.
Actually many tests are linked against libcurl (if available), but a few tests use curl binary (if available).
As tests cannot be executed (in general) when cross-compiling I suggest to put `net-misc/curl` to BDEPEND only to avoid pulling unneeded dependency for binary package installation.

Reply:
We can still do test? ( ... ) in both.